### DISCUSSION TITLE
Missing request object on ws.open?

### DISCUSSION THREAD

**Discussion by jrmgx** [uNetworking/uWebSockets.js#377] [Q&A] [answered]
Hello, first of all thank you for this library it as pleasure to work with.

I have a question about the API:
I can see that per the doc, the handler `open(ws)` only takes one argument: `ws`
but in the underliying lib we can see that `req` is also passed
https://github.com/uNetworking/uWebSockets/blob/master/misc/READMORE.md#the-appws-route
and also on some test files on uWebSockets.js it is also passed 
https://github.com/uNetworking/uWebSockets.js/blob/ea8d4c784d2eeb70ac6884c30b68b17070118d21/tests/Autobahn.js#L25
So I guess, either it has been the case in the past and changed, or it's missing from the underliying API?

Why am I asking this?
I'd like to open a websocket that is aware of the "path" it connected from
something like
```javascript
const app = uWS.App().ws('/:identifier', {
    open: (ws, req) => {
        console.log('A WebSocket connected!');
        console.log('request identifier: ' + req.param[0]); // pseudo code
    },
```

Let me know the best way to do it

**Comment by jrmgx** [comment, marked as answer]
ok found the trick
i have to use the `upgrade` handler

https://unetworking.github.io/uWebSockets.js/generated/interfaces/httpresponse.html#upgrade

**Comment by ghost** [comment]
You're right, the user manual hasn't been updated - will fix
